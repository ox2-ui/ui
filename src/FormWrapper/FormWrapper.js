import React, { PropTypes } from 'react';
import Scroller from '@ox2/scroller/Scroller';
import Button from '@ox2/button/Button';

/**
 * FormWrapper Component
 */
const FormWrapper = ({ children, dirty, submitSucceeded, handleSubmit, promptWidget, pristine, reset, submitting }) => (
  <div className="pos:absolute-0 overflow:hidden h:100p">
    { promptWidget &&
      (promptWidget({ dirty, submitSucceeded }))
    }
    <form onSubmit={handleSubmit}>
      <div className="pos:absolute left:0 right:0 z:2 display:flex flex:items-center padding-x:7 flex:content-between border-color:neutral-faded border:bottom border:2" style={{ height: '44px' }}>
        <Button
          btn="action small"
          type="submit"
          disabled={pristine || submitting}
        >Save</Button>
        <Button
          btn="neutral-dim small"
          type="button"
          disabled={pristine || submitting}
          onClick={reset}
        >Reset</Button>
      </div>
      <Scroller className="padding-x:20" style={{ top: '44px' }}>
        {children}
      </Scroller>
    </form>
  </div>
);


FormWrapper.propTypes = {
  children: PropTypes.node.isRequired,
  dirty: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  promptWidget: PropTypes.func,
  reset: PropTypes.func.isRequired,
  submitSucceeded: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
};

export default FormWrapper;
