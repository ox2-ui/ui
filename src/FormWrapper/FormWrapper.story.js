import React from 'react';
import { storiesOf } from '@kadira/storybook';
import FormWrapper from './FormWrapper';

const handler = () => console.log('🍇'); // eslint-disable-line no-console

storiesOf('FormWrapper', module)
  .add('default', () => (
    <FormWrapper
      dirty={false}
      submitSucceeded={false}
      submitting={false}
      pristine={true}
      reset={handler}
      handleSubmit={handler}
    >
      <div style={{marginTop: '20px'}}>Children</div>
    </FormWrapper>
  ));
