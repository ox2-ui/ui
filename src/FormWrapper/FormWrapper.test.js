/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import FormWrapper from './FormWrapper';

const handler = () => true;

const promptWidget = () => <div>PromptWidget</div>;

it('renders correctly', () => {
  const wrapper = mount(
    <FormWrapper
      dirty={false}
      submitSucceeded={false}
      submitting={false}
      pristine={true}
      reset={handler}
      handleSubmit={handler}
      promptWidget={promptWidget}
    >
      <div>Children</div>
    </FormWrapper>
  );

  expect(wrapper).toMatchSnapshot();
});
