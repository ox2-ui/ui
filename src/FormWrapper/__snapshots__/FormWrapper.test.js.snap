exports[`test renders correctly 1`] = `
<FormWrapper
  dirty={false}
  handleSubmit={[Function]}
  pristine={true}
  promptWidget={[Function]}
  reset={[Function]}
  submitSucceeded={false}
  submitting={false}>
  <div
    className="pos:absolute-0 overflow:hidden h:100p">
    <div>
      PromptWidget
    </div>
    <form
      onSubmit={[Function]}>
      <div
        className="display:flex flex:items-center padding-x:7 flex:content-between border-color:neutral-faded border:bottom border:2"
        style={
          Object {
            "height": "44px",
          }
        }>
        <Button
          btn="action small"
          disabled={true}
          type="submit">
          <button
            data-btn="action small"
            disabled={true}
            type="submit">
            Save
          </button>
        </Button>
        <Button
          btn="neutral-dim small"
          disabled={true}
          onClick={[Function]}
          type="button">
          <button
            data-btn="neutral-dim small"
            disabled={true}
            onClick={[Function]}
            type="button">
            Reset
          </button>
        </Button>
      </div>
      <Scroller
        className="padding-x:20"
        style={
          Object {
            "top": "44px",
          }
        }>
        <div
          className="padding-x:20"
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "backgroundColor": "transparent",
              "bottom": "0",
              "left": "0",
              "overflowX": "hidden",
              "overflowY": "auto",
              "paddingBottom": "40px",
              "position": "absolute",
              "right": "0",
              "top": "44px",
            }
          }>
          <div>
            Children
          </div>
        </div>
      </Scroller>
    </form>
  </div>
</FormWrapper>
`;
